coriolis/tests/osmorphing/test_manager.py

```diff
@@ -0,0 +1,351 @@
+# Copyright 2024 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+import logging
+from unittest import mock
+
+from coriolis import exception
+from coriolis.osmorphing import base as base_osmorphing
+from coriolis.osmorphing import manager
+from coriolis.tests import test_base
+
+
+class ManagerTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis manager module."""
+
+    def setUp(self):
+        super(ManagerTestCase, self).setUp()
+        self.osmorphing_info = {
+            'os_type': 'linux',
+            'ignore_devices': []
+        }
+        self.provider = mock.MagicMock()
+        self.event_handler = mock.MagicMock()
+        self.os_mount_tools = mock.MagicMock()
+        self.event_manager = mock.MagicMock()
+        self.destination_provider = mock.MagicMock()
+        self.worker_connection = mock.MagicMock()
+        self.detected_os_info = mock.MagicMock()
+
+    @mock.patch('coriolis.osmorphing.manager.CONF')
+    def test_get_proxy_settings(self, mock_CONF):
+        mock_CONF.proxy.url = mock.sentinel.proxy_url
+        mock_CONF.proxy.username = mock.sentinel.proxy_username
+        mock_CONF.proxy.password = mock.sentinel.proxy_password
+        mock_CONF.proxy.no_proxy = mock.sentinel.proxy_no_proxy
+
+        expected_result = {
+            "url": mock.sentinel.proxy_url,
+            "username": mock.sentinel.proxy_username,
+            "password": mock.sentinel.proxy_password,
+            "no_proxy": mock.sentinel.proxy_no_proxy,
+        }
+
+        result = manager._get_proxy_settings()
+        self.assertEqual(result, expected_result)
+
+    @mock.patch.object(manager.osdetect_manager, 'detect_os')
+    @mock.patch.object(manager.schemas, 'validate_value')
```

Comment:
Avoid mocking `validate_value` as well.

By setting the following return value on `mock_detect_os`:
```python
mock_detect_os.return_value = {
    "os_type": "linux",
    "distribution_name": "Ubuntu",
    "release_version": "22.04",
    "friendly_release_name": "Ubuntu 22.04",
}
```

the `schemas.validate_value` passes without any problem because it respects the following JSON schema:
https://github.com/cloudbase/coriolis/blob/02123ff7b86c18bca708bca0006872bca7bfa55c/coriolis/schemas/detected_os_morphing_info_schema.json#L1-L24